forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth
+
+import android.content.Context
+import okhttp3.Cookie
+import okhttp3.CookieJar
+import okhttp3.HttpUrl
+import okhttp3.internal.toImmutableList
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+
+class SecureCookieJar(context: Context?,
+                      singleSignOnManager: SingleSignOnManager?,
+                      cacheIntervalMillis: Long?) : CookieJar, OkHttpCookieInterceptor {
+
+    private val singleSignOnManager: SingleSignOnManager
+    private val cacheRef = AtomicReference<MutableSet<Cookie>?>()
+    private val cacheIntervalMillis: Long
+    private val cookieMarshaller = CookieMarshaller()
+
+    init {
+        this.singleSignOnManager = singleSignOnManager ?: Config.getInstance().singleSignOnManager
+        this.cacheIntervalMillis = cacheIntervalMillis
+            ?: (context!!.resources.getInteger(R.integer.forgerock_cookie_cache) * 1000).toLong()
```

Comment:
replace bang bang to optional
 
this.cacheIntervalMillis = cacheIntervalMillis
            ?: ((context?.resources?.getInteger(R.integer.forgerock_cookie_cache) ?: 0) * 1000).toLong()

Reply:
I changed the context to non-null.